web/react/utils/async_client.jsx

```diff
@@ -40,88 +40,48 @@ function isCallInProgress(callName) {
     return true;
 }
 
-export function getChannels(force, updateLastViewed, checkVersion) {
-    var channels = ChannelStore.getAll();
-
-    if (channels.length === 0 || force) {
-        if (isCallInProgress('getChannels')) {
-            return;
-        }
-
-        callTracker.getChannels = utils.getTimestamp();
+export function getChannels(checkVersion) {
+    if (isCallInProgress('getChannels')) {
+        return;
+    }
 
-        client.getChannels(
-            (data, textStatus, xhr) => {
-                callTracker.getChannels = 0;
+    callTracker.getChannels = utils.getTimestamp();
 
-                if (checkVersion) {
-                    var serverVersion = xhr.getResponseHeader('X-Version-ID');
+    client.getChannels(
+        (data, textStatus, xhr) => {
+            callTracker.getChannels = 0;
 
-                    if (!BrowserStore.getLastServerVersion()) {
-                        BrowserStore.setLastServerVersion(serverVersion);
-                    }
+            if (checkVersion) {
+                var serverVersion = xhr.getResponseHeader('X-Version-ID');
 
-                    if (serverVersion !== BrowserStore.getLastServerVersion()) {
-                        BrowserStore.setLastServerVersion(serverVersion);
-                        window.location.reload(true);
-                        console.log('Detected version update refreshing the page'); //eslint-disable-line no-console
-                    }
+                if (serverVersion !== BrowserStore.getLastServerVersion()) {
+                    BrowserStore.setLastServerVersion(serverVersion);
+                    window.location.reload(true);
+                    console.log('Detected version update refreshing the page'); //eslint-disable-line no-console
                 }
 
-                if (xhr.status === 304 || !data) {
-                    return;
+                if (serverVersion !== BrowserStore.getLastServerVersion()) {
+                    BrowserStore.setLastServerVersion(serverVersion);
+                    window.location.href = window.location.href;
+                    console.log('Detected version update refreshing the page'); //eslint-disable-line no-console
```

Comment:
Why do we have two if blocks here? `window.location.href = window.location.href;` could just be moved into the above one after the `window.location.reload` line